### PR TITLE
Open curriculum page to unauthorized users

### PR DESCRIPTION
**Pull request by Ulisseus**
This PR mostly about UX, the idea is to make new user experience as frictionless as possible. 

Currently when new user clicks big GET STARTED button on the landing page he is being redirected to  notion [link](https://www.notion.so/Table-of-Contents-a83980f81560429faca3821a9af8a5e2), then he needs to create an account and only then he is being able to view the curriculum page. First, book notion link seems to be slightly out of date (did c0d3 start as collection of notion lessons?), second there is no reason to do it. This PR takes him directly to curriculum page. 

Navigation bar was changed too. Right now it displays `Home Learning Process Resources Help Contributors ` buttons for unauthorized users. Home is redundant since C0D3 logo already redirects to home. Learning process simply scrolls down a bit. I changed it to authorized users navbar which consists of `Curriculum Repo Journey Help Contributors`. Journey and Resources buttons redirect to the same notion-book link (should we rename them to something more fitting?). 

`review/[lesson]` page now checks for session (unauthorized should be redirected to login page) and for completion of this lesson (you should not see review page if you haven't passed this lesson yet). Test for  `curriculum/[lesson]` was moved from `__tests/__` to `__tests__/curriculum` to match pages folder hierarchy.

### CODE REVIEW

File: components/LandingPage.tsx [hunk 1]
```diff
@@ -34,9 +34,8 @@ const LandingPage: React.FC = () => {
         <div className="row mt-5">
           <div className="col-md-6 offset-md-3 mb-5">
             <NavLink
-              path="https://www.c0d3.com/book"
+              path="https://www.c0d3.com/curriculum"
```
Comment (ggwadera) [inline components/LandingPage.tsx:37]: Use relative path `/curriculum`.

File: components/LandingPage.tsx [hunk 2]
```diff
@@ -228,9 +227,8 @@ const LandingPage: React.FC = () => {
           Become a full stack software engineer - 100% Free!
         </p>
         <NavLink
-          path="https://www.c0d3.com/book"
+          path="https://www.c0d3.com/curriculum"
```
Comment (ggwadera) [inline components/LandingPage.tsx:230]: Use relative path `/curriculum`.

**Review by ggwadera — CHANGES_REQUESTED** [review]
Also the `Curriculum` link on the navbar is styled as active even on the landing page, but it's an issue for another PR because it needs to be fixed on the `NavLink` component to make it use the router to get the current path.

Nice work @Ulisseus 🙂

File: components/LandingPage.tsx (hunk 1 above)
Comment (Ulisseus) [reply components/LandingPage.tsx]: Fixed.

File: components/LandingPage.tsx (hunk 2 above)
Comment (Ulisseus) [reply components/LandingPage.tsx]: Fixed.

**Review by ggwadera — APPROVED** [review]

**Review by anthonykhoa — APPROVED** [review]

**Pull request merged**